utils/usage/visibility_system_manager.go

```diff
@@ -0,0 +1,68 @@
+package usage
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+)
+
+type VisibilitySystemManager struct {
+	serverDetails *config.ServerDetails
+}
+
+func NewVisibilitySystemManager(serverDetails *config.ServerDetails) *VisibilitySystemManager {
+	return &VisibilitySystemManager{
+		serverDetails: serverDetails,
+	}
+}
+
+type labels struct {
+	ProductID                            string `json:"product_id"`
+	FeatureID                            string `json:"feature_id"`
+	OIDCUsed                             string `json:"oidc_used"`
+	JobID                                string `json:"job_id"`
+	RunID                                string `json:"run_id"`
+	GitRepo                              string `json:"git_repo"`
+	GhTokenForCodeScanningAlertsProvided string `json:"gh_token_for_code_scanning_alerts_provided"`
+}
+
+type visibilityMetric struct {
+	Value       int    `json:"value"`
+	MetricsName string `json:"metrics_name"`
+	Labels      labels `json:"labels"`
+}
```

Comment:
**without** `omitempty` it will send here `Value 0` for empty value for example. If its a possible value a believe that its not good. same goes with empty strings.

**with** omitempty , it won't but it won't send actual 0 value if needed.

so it's need to be decided according to context.

Reply:
Thanks @sverdlov93.
Omit empty won't be good in the case for the following reasons:
1. We'd like to send empty strings in the JSON payload, if no  values are sent. I believe that the payload structure is expected to be consistent.
2. The value that we sent is always 1, so it'll never be empty.